src/PaymentChannel.js

```diff
@@ -0,0 +1,114 @@
+import BigNumber from 'bignumber.js';
+
+import paymentChannelStateMessages from './payment_channel_state_service_pb';
+
+export default class PaymentChannel {
+  constructor(channelId, web3, account, paymentChannelStateServiceClient, mpeContract) {
+    this._channelId = channelId;
+    this._web3 = web3;
+    this._account = account;
+    this._mpeContract = mpeContract;
+    this._paymentChannelStateServiceClient = paymentChannelStateServiceClient;
+    this._state = {
+      nonce: new BigNumber(0),
+      lastSignedAmount: new BigNumber(0),
+    };
+  }
+
+  get channelId() {
+    return this._channelId;
+  }
+
+  get nonce() {
+    return this.state.nonce;
+  }
+
+  get lastSignedAmount() {
+    return this.state.lastSignedAmount;
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  updateState(gstate) {
+    this._state = {
+      ...this._state,
+      ...state,
+    }
+  }
+
+  hasSufficientFunds(amount) {
```

Comment:
This specific method could be moved into `PaymentChannelManagementStrategy`.

Reply:
Added a generic answer [here](https://github.com/singnet/snet-sdk-js/pull/1#discussion_r284151536)

Reply:
Addressed in [commit](https://github.com/singnet/snet-sdk-js/pull/1/commits/1adf2e3c86a29e3437e5216bb44a22f035277728)